### PR TITLE
fix(tool-postgresql): Non-existent /workspace dir

### LINKED ISSUE #705: The `gitpod/workspace-postgres` workspace image does not start the PostgreSQL database
### Bug description

Instead of starting the PostgreSQL database, workspaces based on the `gitpod/workspace-postgres` image ([source](https://github.com/gitpod-io/workspace-images/blob/master/chunks/tool-postgresql/Dockerfile)) display the following errors in the terminal:

```
 HISTFILE=/workspace/.gitpod/cmd-0 history -r; {
[ -r /workspace/.prebuild-log-0 ] && cat /workspace/.prebuild-log-0; [ -r /workspace/.gitpod/prebuild-log-0 ] && cat /workspace/.gitpod/prebuild-log-0; true
} && {
echo ""
}
pg_ctl: directory "/workspace/.pgsql/data" does not exist
2022-02-19 15:27:38.932 UTC [579] FATAL:  could not create file "pg_wal/xlogtemp.579": No such file or directory
child process exited with exit code 1
initdb: removing contents of data directory "/workspace/.pgsql/data"
pg_ctl: directory "/workspace/.pgsql/data" is not a database cluster directory
gitpod /workspace/gitpod-test-postgresql $  HISTFILE=/workspace/.gitpod/cmd-0 history -r; {
> [ -r /workspace/.prebuild-log-0 ] && cat /workspace/.prebuild-log-0; [ -r /workspace/.gitpod/prebuild-log-0 ] && cat /workspace/.gitpod/prebuild-log-0; true
> } && {
> echo ""
> }
```

### Steps to reproduce

1. Create a new repository in GitHub
2. Commit a `.gitpod.yml` file with the following content
    ```yaml
    image: gitpod/workspace-postgres

    tasks:
      - command: echo ""
    ```
3. Connect the project to Gitpod via https://gitpod.io/new
4. Start a new workspace

### Workspace affected

_No response_

### Expected behavior

The terminal shows no errors, the PostgreSQL database starts and I can connect to it with `psql`.

### Example repository

https://github.com/mikenikles/gitpod-test-postgresql

### Anything else?

You can verify that the database is not running by trying to connect to it with `psql`.

### PR DESCRIPTION
**Pull request by axonasif**
## Description

`/workspace` dir is non-existent and will get overlapped by gitpod bind mounts on `/workspace` during workspace start.

## Related Issue(s)

Fixes #705 

## How to test

https://gitpod.io/#https://github.com/axonasif/test/tree/postgresql-test

## Release Notes

```release-note
```

### CODE REVIEW

**Comment by csweichel** [comment]
That's a good observation. The original intent was to place the data files in `/workspace` so that they're still available after a workspace restart. Can we find a way to keep that functionality?

**Comment by axonasif** [comment]
> That's a good observation. The original intent was to place the data files in `/workspace` so that they're still available after a workspace restart. Can we find a way to keep that functionality?

It's persisted on workspace restart.

**Comment by mikenikles** [comment]
Do I understand this correctly @axonasif, you're saying the functionality Chris asks to keep is already kept and there are no additional changes needed to merge this PR?

**Comment by axonasif** [comment]
> Do I understand this correctly @axonasif, you're saying the functionality Chris asks to keep is already kept and there are no additional changes needed to merge this PR?

Yeah, same functionality. But I still need to fix the dazzle test for this chunk.

**Comment by kylos101** [comment]
Hey @axonasif , [the test is currently failing](https://github.com/gitpod-io/workspace-images/runs/5272157470?check_suite_focus=true#step:10:15975) for this PR. 

Short term, this will help you iterate more quickly, to easily run tests:
1. Run this `for file in $(ls chunks) ; do dazzle project ignore "$file"; done`
2. Remove `tool-postgresql` from the ignore list in the dazzle.yaml file at the bottom
3. Edit `

Long term we'll [make an improvement](https://github.com/gitpod-io/dazzle/issues/38) to the `dazzle-up.sh`.

**Comment by kylos101** [comment]
@axonasif slight [edit](https://github.com/gitpod-io/workspace-images/pull/706#issuecomment-1059248013) (above).

**Comment by axonasif** [comment]
> 

That seems like a good in-depth way. I was moving out all other chunks from `chunks/` and only keeping the one I was working on(previously was moving out `tests/`). Is this approach followable too?

**Comment by kylos101** [comment]
> > 
> 
> That seems like a good in-depth way. I was moving out all other chunks from `chunks/` and only keeping the one I was working on(previously was moving out `tests/`). Is this approach followable too?

Yep! I've done that too. :wink: Basically removing the chunks I don't want, and the combos from the yaml file, and then proceeding. And when done, putting them back.

**Comment by axonasif** [comment]
> > > 
> > 
> > 
> > That seems like a good in-depth way. I was moving out all other chunks from `chunks/` and only keeping the one I was working on(previously was moving out `tests/`). Is this approach followable too?
> 
> Yep! I've done that too. 😉 Basically removing the chunks I don't want, and the combos from the yaml file, and then proceeding. And when done, putting them back.

👀 So I was experimenting with this method and found that if you specify the chunk tag/version too (e.g `lang-go:1.17.5`) then dazzle doesn't actually ignore, rather it does something random.
```yaml
ignore:
  - lang-go
``` works

I ended up automating the process somewhat. You can try it here https://github.com/axonasif/workspace-images/blob/axon/dazzle-up-patch/dazzle-up.sh if you want 😉 

Testing:
- Ignore all and only build the specified ones: `./dazzle-up.sh -i all lang-node:17.4.0`
- Ignore a few and build a few: `./dazzle-up.sh --ignore=lang-java,lang-go lang-node lang-c`

Note: `-i`, `--ignore=`, `--ignore` are the same option.

**Comment by kylos101** [comment]
Thanks, @axonasif ! I know @Furisto plans to make an improvement in #38. 

For posterity, @Furisto , check out what @axonasif did [here](https://github.com/gitpod-io/workspace-images/pull/706#issuecomment-1059545468).

**Comment by kylos101** [comment]
@axonasif can you peek at the test failure in this PR?

**Comment by axonasif** [comment]
> @axonasif can you peek at the test failure in this PR?
> 
> edit: this assumes you can see the failures from your Gitpod workspace, and execute the tests more easily there.

Yeah, I'm doing that now 😄 
Another feedback that I wanted to share with you too: https://github.com/gitpod-io/workspace-images/pull/587/files#r806870960

**Comment by axonasif** [comment]
@kylos101 the tests were fixed 😄

**Comment by kylos101** [comment]
Thanks @axonasif ! Looks like there is some prebuild trouble. Can you ping us once the pull request action is successful, or let us know if you have trouble?

**Comment by axonasif** [comment]
> Thanks @axonasif ! Looks like there is some prebuild trouble. Can you ping us once the pull request action is successful, or let us know if you have trouble?

Oh sorry, I should have had installed pre-commit on my workspace, did that and now everything's fine 😅

File: tests/tool-postgresql.yaml
Comment (kylos101) [inline tests/tool-postgresql.yaml:10]: It seems wrong to me, to remove this test, without adding a new one.

May I ask you to add a test, asserting that the underlying data files for the database exist?

**Review by kylos101 — CHANGES_REQUESTED** [review]
Hey @axonasif , may I ask you to add a new test (to replace the one you're deleting)? Or, share why you think it is unnecessary?

File: tests/tool-postgresql.yaml
Comment (axonasif) [reply tests/tool-postgresql.yaml:10]: Hey @kylos101, I feel that this test is unnecessary since `pg_ctl` does that check anyways.
image

`pg_ctl` is invoked from the `.bashrc.d` hook
https://github.com/gitpod-io/workspace-images/blob/8f84ceddab51fa26d4f5ff53dd23b27acb929e9e/tests/tool-postgresql.yaml#L8

**Comment by mikenikles** [comment]
This PR currently blocks me from enabling Gitpod for the RedwoodJS starter project at https://github.com/redwoodjs/gitpod-starter.

I see the most recent event was @axonasif asking for a review from @kylos101. Is this accurate? What can I do to help get this PR merged and fix PostgreSQL support on Gitpod?

**Comment by axonasif** [comment]
Hey @mikenikles, now I'm waiting for another review and willing to make any changes if necessary after that as well 🙏

**Review by kylos101 — APPROVED** [review]
Amazing work, sorry about the delay @axonasif ! This should produce a new version that has a timestamp tag in Docker in roughly 1 hour.

**Note**: it won't have the latest tag until our next workspace-cluster deployment. But, users can always use the timestamp tagged image.

**Pull request merged**
